app/scripts/models/reliers/oauth.js

```diff
@@ -140,10 +146,10 @@ define(function (require, exports, module) {
       self._importRequiredSearchParam('client_id', 'clientId');
       self._importRequiredSearchParam('scope', 'scope');
       self.importSearchParam('state');
-      self.importSearchParam('redirect_uri', 'redirectUri');
+      self._importValidateOptionalSearchParam('redirect_uri', 'redirectUri', Validate.isUriValid, OAuthErrors);
```

Comment:
It seems like I need to make the validation functionality a little bit more flexible so that it can be used for more than just search params. Something like [Joi](https://github.com/hapijs/joi) but maybe more compact. Or it might be possible to extend [isDataValid](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/lib/validate.js#L125) to support testing for values?